### ISSUE TITLE
Change license to AGPLv3+

### ISSUE THREAD

**Issue by pizzamaker** [HelloZeroNet/ZeroNet#224]
Affero further protects the project against SaaSS :)

**Comment by HelloZeroNet** [comment]
Well, I'm not against if someone offers ZeroNet hosting as service.

**Comment by pizzamaker** [comment]
Me neither, but this assures you that the service provider will need to publish your source code freely and any modifications made to it :)

**Comment by unsystemizer** [comment]
GPL is restrictive enough. It seems as if pizzamaker is trying to ensure noone tries to use the s/w.

**Comment by pizzamaker** [comment]
@unsystemizer AGPL does not restrict commercial use of software. It merely posits that any changes to the code must be given back to the community that made it.

**Comment by unsystemizer** [comment]
@pizzamaker yeah and what kind of effect does that have on commercial use of software? 
Or, if AGPL doesn't restrict anything more than the current license, why are you proposing it? Of course you're proposing it to restrict the use. 
But I don't want to get into license wars here, I'll just say take a look at the recent trend (I won't copy the graphic here, but pay attention to Figure 2 and the collapse in GPL. And the explosive growth of permissive licenses such as MIT and Apache. 

Source: http://osswatch.jiscinvolve.org/wp/2015/02/05/open-source-software-licensing-trends/

**Comment by pizzamaker** [comment]
Affero closes a known loophole in the GPL, when it comes to webservices: 

> Both versions of the Affero GPL were designed to close a perceived application service provider (ASP) loophole in the ordinary GPL, where, by using but not distributing the software, the copyleft provisions are not triggered. Each version differs from the version of the GNU GPL on which it is based in having an added provision addressing use of software over a computer network. This provision requires that the full source code be made available to any network user of the AGPL-licensed work, typically a web application.
https://en.wikipedia.org/wiki/Affero_General_Public_License

**Comment by bashrc** [comment]
It's amusing to consider whether ZeroNet is a client or server application. It's neither and both. As I understand it, AGPL is primarily useful for server based applications which the user doesn't install themselves - similar to Google's stuff which might be free in the background but unfree from the user point of view. I don't have a strong opinion on whether it should be GPL or AGPL. If there are use cases where ZeroNet could be run as a server then AGPL would make sense.

**Comment by pizzamaker** [comment]
@bashrc Due to the SaaSS trend, it's best to use AGPL for everything these days, since even full-featured desktop applications like GIMP have been transformed into web apps.

**Comment by deavmi** [comment]
I agree that this would be a good license.

**Comment by unsystemizer** [comment]
nonsense (and poorly argued one for at that)
above i posted a link that shows what the real trend is.

**Comment by deavmi** [comment]
K. I would use it though. That's my opinion. Not forcing anyone to use it.

On 06 June 2016 7:40:24 AM SAST, unsystemizer <notifications@github.com> wrote:
>nonsense (and poorly argued one for at that)
>above i posted a link that shows what the real trend is.
>
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/HelloZeroNet/ZeroNet/issues/224#issuecomment-223872586

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by unsystemizer** [comment]
i said the same (not demanding anything), but since you are arguing for a change can you provide arguments as you see them?

**Comment by deavmi** [comment]
Well someone can host a proxy and therefore using ZeroNet as a server.

**Comment by unsystemizer** [comment]
so what? thats the whole point, to let people share information. 
the same setup can be done with Bittorrent and IPFS and tons of other apps.

**Comment by deavmi** [comment]
Yeah. But they could make improvements to the proxy they run that you might like. Intact they could improve the whole ZeroNet system.

On 06 June 2016 8:58:57 PM SAST, unsystemizer <notifications@github.com> wrote:
>so what? thats the whole point, to let people share information. 
>the same setup can be done with Bittorrent and IPFS and tons of other
>apps.
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/HelloZeroNet/ZeroNet/issues/224#issuecomment-224054274

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by unsystemizer** [comment]
in all likelihood we would benefit from that

**Comment by deavmi** [comment]
Yeah. My point exactly.

On 06 June 2016 9:18:39 PM SAST, unsystemizer <notifications@github.com> wrote:
>in all likelihood we would benefit from that
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/HelloZeroNet/ZeroNet/issues/224#issuecomment-224059531

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by KindlyFire** [comment]
Has anybody looked at the [Apache License](http://choosealicense.com/licenses/apache-2.0/) ? Looks like a nice fit to me.

**Comment by bignose-debian** [comment]
> Well, I'm not against if someone offers ZeroNet hosting as service.

Yes, that would be great!

The issue though, is to ensure the users of that service can get the ZeroNet source code that's being used, and improve and share it. That is why the AGPL 3+ is a good choice of license.
